### ISSUE TITLE
Illegal access to normal neighbor of halo vertex

### ISSUE THREAD

**Issue by maxaehle** [su2code/SU2#1430]
**Describe the bug**
In CFVMFlowSolverBase<>::Friction_Forces, there is the following piece of code ([permalink](https://github.com/su2code/SU2/blob/641d2544d8958e9c74fbe9e22d9eb16cbc99167f/SU2_CFD/include/solvers/CFVMFlowSolverBase.inl#L2477)):
```c++
    /*--- Loop over the vertices to compute the forces ---*/
    
    for (iVertex = 0; iVertex < geometry->nVertex[iMarker]; iVertex++) {
      iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
      iPointNormal = geometry->vertex[iMarker][iVertex]->GetNormal_Neighbor();
      ...
    }
```
However, a normal neighbor exists only for vertices on the domain (see [`CMultiGridGeometry::FindNormal_Neighbor`](https://github.com/su2code/SU2/blob/5655d91d7da192ad0b2a231f9ab5334c2235c10a/Common/src/geometry/CMultiGridGeometry.cpp#L1159) and [`CPhysicalGeometry::FindNormal_Neighbor`](https://github.com/su2code/SU2/blob/5655d91d7da192ad0b2a231f9ab5334c2235c10a/Common/src/geometry/CPhysicalGeometry.cpp#L8470)), not for halo points.   
**Edit**: The link for `CPhysicalGeometry` was wrong. `CPhysicalGeometry` actually computes the normal neighbors for halo points, `CMultigridGeometry` doesn't. 

`iPointNormal` is needed to retrieve `Coord_Normal`, which is needed to compute `WallDist`/`WallDistMod`, which are needed to compute `YPlus`. To fix the above problem, I would therefore suggest to compute `YPlus` only for domain points.

Looking for locations where `YPlus` could be accessed (to see whether `YPlus` is needed on non-domain points), I found e.g. [`CTurbSSTSolver::SetTurbVars_WF`](https://github.com/su2code/SU2/blob/5655d91d7da192ad0b2a231f9ab5334c2235c10a/SU2_CFD/src/solvers/CTurbSSTSolver.cpp#L494). However there again the normal neighbor is accessed without checking that we're on a domain point, so likely this should also be changed.

**Comment by pcarruscag** [comment]
Have you checked if you ever get halo point in normal markers?

**Comment by maxaehle** [comment]
Yes, in the testcase described here: https://github.com/su2code/SU2/pull/1362#issuecomment-964163647 I saw in the debugger that halo vertices occur in the loop.

**Comment by pcarruscag** [comment]
Right, in the viscous forces case, it should be ok to move the "if Domain" up and avoiding computations on halo points.

**Comment by pcarruscag** [comment]
I guess this is a problem for FULL MG only, because CPhysicalGeometry finds normal neighbors also for halo points.

**Comment by pcarruscag** [comment]
@bigfooted , I see we are skiping halo nodes when we compute TauWall via wall functions, probably also because of this PointNormal problem?
This may be a problem, because some edges might be missing one tau wall when computing the viscous fluxes.
If that is the case the easier solution is to communicate TauWall.

**Comment by maxaehle** [comment]
By the way, is there a reason why `CMultiGridGeometry::FindNormal_Neighbor` does not compute the normal neighbor for halo nodes (if anyone happens to know)?
![screenshot](https://user-images.githubusercontent.com/72806890/141294574-e7a2a939-8949-4324-a6cb-ddce59ed6a4b.png)

**Comment by maxaehle** [comment]
> in the viscous forces case, it should be ok to move the "if Domain" up and avoiding computations on halo points.

Corrected in commit f6f986ec256a1c47199703ef6540145506dbbdc3 in #1362 similar to that, moving the code depending on the nearest neighbor access (i.e. YPlus calculation) down into an "if Domain" block. Note however that after this change YPlus is not computed on halo cells any more, which might be problematic e.g. for [CTurbSSTSolver::SetTurbVars_WF](https://github.com/su2code/SU2/blob/5655d91d7da192ad0b2a231f9ab5334c2235c10a/SU2_CFD/src/solvers/CTurbSSTSolver.cpp#L494).
